react/src/components/UserNodeList.tsx

```diff
@@ -374,7 +411,158 @@ const UserNodeList: React.FC<UserNodeListProps> = () => {
+          },
+          {
+            key: 'container_uid',
+            title: 'Container UID',
+            render: (__, record) =>
+              record.container_uid ? (
+                <BAIText ellipsis style={{ maxWidth: 200 }}>
+                  {record.container_uid}
+                </BAIText>
+              ) : (
+                '-'
+              ),
+          },
+          {
+            key: 'container_main_gid',
+            title: 'Container Main GID',
+            render: (__, record) =>
+              record.container_main_gid ? (
+                <BAIText ellipsis style={{ maxWidth: 200 }}>
+                  {record.container_main_gid}
+                </BAIText>
+              ) : (
+                '-'
+              ),
+          },
+          {
+            key: 'container_gids',
+            title: 'Container GIDs',
+            render: (__, record) =>
+              !_.isEmpty(record.container_gids) ? (
+                <BAIText ellipsis style={{ maxWidth: 200 }}>
+                  {_.join(record.container_gids, ', ')}
+                </BAIText>
+              ) : (
+                '-'
+              ),
+          },
+          {
+            key: 'description',
+            title: t('credential.Description'),
+            dataIndex: 'description',
+          },
+          {
+            key: 'sudo_session_enabled',
+            title: 'Sudo Session Enabled',
+            sorter: true,
+            render: (_, record) => (
+              <BooleanTag value={record.sudo_session_enabled} />
+            ),
+          },
+          {
+            key: 'need_password_change',
+            title: 'Need Password Change',
+            sorter: true,
+            render: (_, record) => (
+              <BooleanTag value={record.need_password_change} />
+            ),
+          },
+          {
+            key: 'totp_activated',
+            title: '2FA',
+            sorter: true,
+            render: (_, record) => (
+              <BooleanTag
+                value={record.totp_activated}
+                trueLabel={t('general.Enabled')}
+                falseLabel={t('general.Disabled')}
+              />
+            ),
+          },
+          {
+            key: 'status_info',
+            title: t('credential.StatusInfo'),
+            dataIndex: 'status_info',
+          },
+          {
+            title: t('general.CreatedAt'),
+            render: (__, record) => dayjs(record.created_at).format('lll'),
+            sorter: true,
+            defaultSortOrder: 'descend',
+          },
+          {
+            title: t('general.ModifiedAt'),
+            render: (__, record) => dayjs(record.modified_at).format('lll'),
+            sorter: true,
+          },
+          queryParams.status !== 'active' && {
+            key: 'status',
+            title: t('credential.Status'),
+            render: (__, record) => record.status,
+            sorter: true,
+          },
+        ] as BAIColumnsType<UserNode>)}
```

Comment:
Using generic types seems safer than using as for type conversion. \<BAITable\<UserNode>

Reply:
Please check BAIUserNodes component in #4863 . I refactored UserNodeList to BAIUserNdoes and UserManagement.